src/HttpExtractor.php

```diff
@@ -21,7 +24,23 @@ public function __construct(
 
     public function extract(UriInterface $httpSource, string $filesystemDestination): void
     {
-        $this->client->get($httpSource, ['sink' => $filesystemDestination]);
+        try {
+            $this->client->get($httpSource, ['sink' => $filesystemDestination]);
+        } catch (ClientException|ServerException $e) {
+            throw new UserException(sprintf(
+                'Server returned HTTP %s for "%s"',
+                $e->getCode(),
+                (string)$httpSource
+            ), 0, $e);
+        } catch (ConnectException $e) {
+            if (strpos($e->getMessage(), 'cURL error 6: Could not resolve host') !== 0) {
```

Comment:
Tak tohle mě totálně zmátlo! Není nějakej lepší způsob jak to zapsat? Už jsem si `var_dump`oval v testech, než mi došlo, že s díváš, jestli `cURL error 6: Could not resolve host` je na začátku stringu. Osobně bych použil 
```
if (strpos($e->getMessage(), 'cURL error 6: Could not resolve host') !== false)
```
což z hlediska kódu není jednoznačný, ale je to možná čitelnější? Nevím! :-)

Reply:
nemelo by ten curl kod byt v  `$e->getHandlerContext()` ?